src/scraper/scraper.cpp

```diff
@@ -4547,6 +4534,11 @@ NN::Superblock ScraperGetSuperblockContract(bool bStoreConvergedStats, bool bCon
                     ConvergedScraperStatsCache.Convergence = StructConvergedManifest;
 
                     superblock = NN::Superblock::FromConvergence(ConvergedScraperStatsCache);
+
+                    if (!IsV11Enabled(nBestHeight)) {
```

Comment:
This may not be threadsafe. nBestHeight is normally protected by cs_main I think. ScraperGetSuperblockContract is called in a number of places in the scraper from the scraper thread, not the "main" thread... so we need a cs_main lock. I get nervous about deadlocks with that. Maybe I am wrong and it is not an issue.

What about converting nBestHeight to an atomic to go lockless?

Reply:
Based on a discussion with @cycy, since an integer can't be corrupted on read on any of our platforms, snooping this is ok, and even if it is slightly out of date, it should be ok.